var/spack/repos/builtin/packages/foam-extend/package.py

```diff
@@ -111,189 +107,77 @@ class FoamExtend(Package):
     depends_on('scotch~metis+mpi', when='+ptscotch')
     depends_on('metis@5:',         when='+metis')
     depends_on('parmetis',         when='+parmetis')
-    depends_on('parmgridgen',      when='+parmgridgen')
+    # mgridgen is statically linked
+    depends_on('parmgridgen',      when='+parmgridgen', type='build')
```

Comment:
Might want to keep this as the default: build/link. Otherwise Spack's compiler wrappers won't automatically add `-L/path/to/mgridgen/lib`.